File: app/src/main/java/com/jsoft/jcomic/praser/DM5EpisodeParser.java
package com.jsoft.jcomic.praser;

import android.util.Log;

import com.jsoft.jcomic.FullscreenActivity;
import com.jsoft.jcomic.helper.EpisodeDTO;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DM5EpisodeParser extends EpisodeParser {
    public DM5EpisodeParser(EpisodeDTO episode, FullscreenActivity activity) {
        super(episode, activity, "UTF-8");
    }

    private String intCode(int i) {
        String s = "";
        String characters="0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
        if (i > characters.length()) {
            s = s + characters.charAt(i/characters.length());
        }
        s = s + characters.charAt(i % characters.length());
        return s;
    }

    protected void getEpisodeFromUrlResult(List<String> result) {
        String s = "";
        for (int i=0; i<result.size();i++) {
            s = s + result.get(i);
        }

        List<String> imageUrlList = new ArrayList<String>();

        Pattern p = Pattern.compile("eval\\(function\\(p,a,c,k,e,d\\)\\{.+\\}\\((.+)\\)\\s*</script>");
        Matcher m = p.matcher(s);
        List<String> allMatches = new ArrayList<String>();
        while (m.find()) {

            String s1= m.group(1);
            Pattern p1 = Pattern.compile("\\d+,\\d+,\'(.*)\'\\.split");
            Matcher m1 = p1.matcher(s1);
            String[] replaceStrArray= new String[0];
            while (m1.find()) {
                String s3=m1.group(1);
                //Log.e("jComics 1", s3);
                replaceStrArray = s3.split("\\|");
            }
            String s2= m.group(1);
            Pattern p2 = Pattern.compile("=\\[(.+?)\\];");
            Matcher m2 = p2.matcher(s2);
            while (m2.find()) {
                String s3=m2.group(1);
                String[] urls = s3.split(",");
                for (int i=0; i<urls.length; i++) {

                    String imgUrl = urls[i];
                    imgUrl = imgUrl.replaceAll("\\\\\\'", "");
                    for (int j=0; j < replaceStrArray.length; j++) {
                        if (replaceStrArray[j].length() > 0) {
                            imgUrl = imgUrl.replaceAll("\\b" + intCode(j) + "\\b", replaceStrArray[j]);
                        }
                    }
                    imageUrlList.add(imgUrl);
                }
            }
        }
        episode.setPageCount(imageUrlList.size());
        episode.setImageUrl(imageUrlList);
        activity.onEpisodeFetched(episode);
    }
}

File: app/src/main/java/com/jsoft/jcomic/praser/DM5EpisodeParser.kt
package com.jsoft.jcomic.praser

import com.jsoft.jcomic.helper.EpisodeDTO
import java.util.*
import java.util.regex.Pattern

class DM5EpisodeParser(episode: EpisodeDTO, listener: EpisodeParserListener) : EpisodeParser(episode, listener, "UTF-8") {

    private fun intCode(i: Int): String {
        var s = ""
        val characters = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
        if (i >= characters.length) {
            s += characters[i / characters.length]
        }
        s += characters[i % characters.length]
        return s
    }

    override fun getEpisodeFromUrlResult(result: List<String>) {
        var s = ""
        for (i in result.indices) {
            s += result[i]
        }

        val imageUrlList = ArrayList<String>()

        val p = Pattern.compile("eval\\(function\\(p,a,c,k,e,d\\)\\{.+\\}\\((.+)\\)\\s*</script>")
        val m = p.matcher(s)
        while (m.find()) {

            val s1 = m.group(1)
            val p1 = Pattern.compile("\\d+,\\d+,\'(.*)\'\\.split")
            val m1 = p1.matcher(s1)
            var replaceStrArray = arrayOfNulls<String>(0)
            while (m1.find()) {
                val s3 = m1.group(1)
                replaceStrArray = s3.split("\\|".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            }

            val s2 = m.group(1)
            val p2 = Pattern.compile("=\\[(.+?)\\];")
            val m2 = p2.matcher(s2)
            while (m2.find()) {
                val s3 = m2.group(1)
                val urls = s3.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                for (url in urls) {
                    var imgUrl = url
                    imgUrl = imgUrl.replace("\\\\\\'".toRegex(), "")
                    for (j in replaceStrArray.indices) {
                        if (replaceStrArray[j]!!.isNotEmpty()) {
                            imgUrl = imgUrl.replace(("\\b" + intCode(j) + "\\b").toRegex(), replaceStrArray[j]!!)
                        }
                    }
                    imageUrlList.add(imgUrl)
                }
            }
        }
        episode.pageCount = imageUrlList.size
        episode.imageUrl = imageUrlList
    }
}
